### PR TITLE
o/devicestate: consider the current model's validation sets when creating a recovery system

### PR DESCRIPTION
**Pull request by andrewphelpsj**
I forgot to consider the current model's validation sets in the first PR. This change makes creating a recovery system respect the validation sets in the current model. The main effect of this change is that you cannot create recovery systems with validation sets that conflict with the validation sets of the current model.

### CODE REVIEW

File: overlord/assertstate/validation_set_tracking.go [hunk 1]
```diff
@@ -208,14 +202,54 @@ func TrackedEnforcedValidationSets(st *state.State, extraVss ...*asserts.Validat
...
+	return sets, nil
+}
+
+// TrackedEnforcedValidationSetsForModel returns a ValidationSets object for
+// currently tracked validation sets that are in enforcing mode and also
+// associated with the specified model.
+func TrackedEnforcedValidationSetsForModel(st *state.State, model *asserts.Model) (*snapasserts.ValidationSets, error) {
+	modelSets := make(map[string]bool, len(model.ValidationSets()))
+	for _, vs := range model.ValidationSets() {
+		modelSets[fmt.Sprintf("%s:%s", vs.AccountID, vs.Name)] = true
```
Comment (bboozzoo) [inline overlord/assertstate/validation_set_tracking.go:224]: there's a `ValidationSetKey` defined near the beginning of this file, could we use it here and in `TrackedEnforcedValidationSets` ?

Comment (andrewphelpsj) [reply overlord/assertstate/validation_set_tracking.go:224]: Good call, thank you!

**Review by bboozzoo — APPROVED** [review]
LGTM

File: overlord/assertstate/validation_set_tracking.go [hunk 2]
```diff
@@ -194,28 +194,62 @@ func ValidationSets(st *state.State) (map[string]*ValidationSetTracking, error)
...
 func TrackedEnforcedValidationSets(st *state.State, extraVss ...*asserts.ValidationSet) (*snapasserts.ValidationSets, error) {
-	valsets, err := ValidationSets(st)
-	if err != nil {
-		return nil, err
-	}
-
-	db := DB(st)
 	sets := snapasserts.NewValidationSets()
 
 	skip := make(map[string]bool, len(extraVss))
```
Comment (pedronis) [inline overlord/assertstate/validation_set_tracking.go:199]: nitpick: my preference would be to call this setsToSkip?

File: overlord/assertstate/validation_set_tracking.go [hunk 3]
```diff
@@ -194,28 +194,62 @@ func ValidationSets(st *state.State) (map[string]*ValidationSetTracking, error)
...
+
+	sets := snapasserts.NewValidationSets()
+	if err := trackedEnforcedValidationSets(st, skipSet, sets); err != nil {
+		return nil, err
+	}
+
+	return sets, nil
+}
+
+func trackedEnforcedValidationSets(st *state.State, skipSet func(string) bool, sets *snapasserts.ValidationSets) error {
```
Comment (pedronis) [inline overlord/assertstate/validation_set_tracking.go:239]: and name here s/skipSet/skip/

**Review by pedronis — APPROVED** [review]
thanks, only a naming nitpick

**Pull request merged**